pkg/lib/cockpit.js

```diff
@@ -30,11 +30,17 @@ import { transport_origin, calculate_application, calculate_url } from './cockpi
 import { Location } from 'cockpit/_internal/location';
 import { ensure_transport, transport_globals } from './cockpit/_internal/transport';
 import { FsInfoClient } from "./cockpit/fsinfo";
+import { fetch_info } from './cockpit/_internal/info';
 
 function factory() {
     const cockpit = { };
     event_mixin(cockpit, { });
 
+    cockpit.init = async () => {
+        await new Promise(resolve => ensure_transport(resolve));
+        Object.assign(cockpit.info, await fetch_info());
```

Comment:
Could this be skipped if `cockpit.info` is already populated? Then you could call `cockpit.init()` multiple times and it would be idempotent and fast. I.e. exactly how `cockpit.user` behaves, which makes it easier to port code to this.

Reply:
Sure.  We could do this.  It wasn't my plan that `.init()` gets called more than once, of course...

Reply:
Indeed I think we shouldn't advertise/encourage this. As `cockpit.ready()` it would make more sense, but you said that this is somehow tricky do to.

Reply:
What would have been tricky is to handle concurrent possible initializations, for example, if it happened implicitly when the transport came online.  Indeed, even in this case, if we "check first and return early" we have no guarantee that we haven't requested the data already and aren't making a concurrent request.

So I can add a case for "return early because we already have the data" but even that wouldn't be a guarantee of only one access.  That would require "more" indeed.

Reply:
I think I'm just going to leave this alone for now.

Reply:
Right, but this would be more of an optimization. Neither os-release nor the user nor the version should change underneath a running session.